spec/rubocop/cop/rspec/avoid_setup_hook_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::AvoidSetupHook do
+  context 'when using blocks' do
+    it 'accepts before call' do
+      expect_no_offenses(<<~RUBY)
+        before do
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+
+    it 'registers an offense on setup call' do
+      expect_offense(<<~RUBY)
+        setup do
+        ^^^^^^^^ Use `before` instead of `setup`.
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+  end
+
+  context 'when autocorrecting offenses' do
+    it 'replaces `setup` with `before`' do
+      expect_offense(<<~RUBY)
+        setup do
+        ^^^^^^^^ Use `before` instead of `setup`.
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        before do
+          allow(foo).to receive(:bar)
+        end
+      RUBY
+    end
+  end
```

Comment:
I believe there's not much of an added value of having this in a separate context.
Just move `expect_correction` to `registers an offense on setup call` and it'll do it.
That is, to my best memory, how most of our specs are written.